cpp/src/bitmask/null_mask.cu

```diff
@@ -372,6 +458,115 @@ cudf::size_type count_unset_bits(bitmask_type const *bitmask, size_type start,
   return (num_bits - detail::count_set_bits(bitmask, start, stop, stream));
 }
 
+std::vector<size_type>
+segmented_count_set_bits(bitmask_type const* bitmask,
+                         std::vector<size_type> const& indices,
+                         cudaStream_t stream) {
+  CUDF_EXPECTS(indices.size() % 2 == 0, "Invalid ranges.");
+
+  if ((bitmask == nullptr) || (indices.size() == 0)) {
+    return std::vector<size_type>{};
+  }
+
+  size_type num_ranges = indices.size() / 2;
+  thrust::host_vector<size_type> h_first_indices(num_ranges);
+  thrust::host_vector<size_type> h_last_indices(num_ranges);
+  for (size_type i = 0; i < num_ranges; i++) {
```

Comment:
Thanks to @codereport for the idea, we can do this with an algorithm!

```
thrust::partition_copy(std::begin(indices), std::end(indices), thrust::make_counting_iterator(0), 
                           h_first_indices.begin(), h_last_indices.begin(), 
                           [](auto i)( auto is_even = [](auto i){return (i%2)==0;}; return is_even(i););
```

Reply:
This is cool!